src/messageFactory.ts

```diff
@@ -26,6 +26,55 @@ export default class MessageFactory {
     } as NewMessage<MessagePayload>);
   }
 
+  /**
+   * Creates the message for chunk
+   *
+   * @template MessagePayload - the type describing structure of the message payload
+   *
+   * @param [payload] - additional data to send
+   */
+  public static createBufferMessage<MessagePayload>(payload?: MessagePayload): string {
```

Comment:
the difference between the `createBufferMessage` and `createForUpload` is still not clear. Do we need these two methods? If so, the naming should be improved